src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedVariantContextUtils.java

```diff
@@ -0,0 +1,283 @@
+     * @return the string version of the attribute value.
+     * @throws IllegalArgumentException if {@code genotype} or {@code key} is {@code null}.
+     * @throws NumberFormatException if the annotation exists, and its string transformation cannot be convert into an integer.
+     */
+    public static int getAttributeAsInt(final Genotype genotype, final String key, final int defaultValue) {
+        Utils.nonNull(genotype);
+        Utils.nonNull(key);
+        final Object value = genotype.getExtendedAttribute(key);
+        if (value == null) {
+            return defaultValue;
+        } else {
+            try {
+                return Integer.parseInt(String.valueOf(value));
+            } catch (final NumberFormatException ex) {
+                throw new NumberFormatException(String.format("attribute '%s' does not have a valid integer value: '%s'", key, String.valueOf(value)));
+            }
+        }
+    }
+
+    /**
+     * Returns an attribute as a double.
+     * @param genotype the source genotype.
+     * @param key the attribute key.
+     * @param defaultValue default value to return in case the attribute does not have a value defined.
+     * @return the string version of the attribute value.
+     * @throws IllegalArgumentException if {@code genotype} or {@code key} is {@code null}.
+     * @throws NumberFormatException if the annotation exists, and its string transformation cannot be convert into a double.
+     */
+    public static double getAttributeAsDouble(final Genotype genotype, final String key, final double defaultValue) {
+        Utils.nonNull(genotype);
+        Utils.nonNull(key);
+        final Object value = genotype.getExtendedAttribute(key);
+        if (value == null) {
+            return defaultValue;
+        } else {
+            try {
+                return Double.parseDouble(String.valueOf(value));
+            } catch (final NumberFormatException ex) {
+                throw new NumberFormatException(String.format("attribute '%s' does not have a valid double value: '%s'", key, String.valueOf(value)));
+            }
+        }
+    }
+
+    /**
+     * Instead of using the GQ value, it re-calculates the quality from the PL so that it does not need
+     * to be bounded by an artificial maximum such as the standard GQ = 99.
+     * @param genotype the PL source genotype.
+     *
+     * @return never {@code null}, but {@link Double#NaN} if there is no PLs.
+     * @throws IllegalArgumentException if {@code genotype} is {@code null}
+     */
+    public static double calculateGenotypeQualityFromPLs(final Genotype genotype) {
+        Utils.nonNull(genotype);
+        if (!genotype.hasPL()) {
+            return Double.NaN;
+        } else {
+            int[] PLs = genotype.getPL();
+            if (PLs.length <= 1) {
+                return Double.NaN;
+            } else {
+                int best = PLs[0], secondBest = Integer.MAX_VALUE;
+                for (int i = 1; i < PLs.length; i++) {
```

Comment:
This motif of best and second best was used earlier in this PR.  Could you extract a method for this?